### DISCUSSION TITLE
Lost openCL after using pocl on MacOs

### DISCUSSION THREAD

**Discussion by yves-surrel** [pyopencl/pyopencl#517] [Troubleshooting] [answered]
I have tried pocl on my Mac mini M1, where I had previously a working install of PyOpenCL.

I removed pocl, but now I get the infamous message `clGetPlatformIDs failed: PLATFORM_NOT_FOUND_KHR`.

How to get back to the previous situation? I hope pocl has not messed up some internals in the OS file system.

**Comment by inducer** [comment]
- How'd you install pyopencl?
- If you set up a fresh conda environment as described in the [docs](https://documen.tician.de/pyopencl/misc.html), do you get a working system?
- You can rest assured that no aspect of pyopencl installation should modify (at least intentionally) your global system. Of course, if you `sudo python setup.py install`, it will attempt to install itself into your system-wide interpreter... but don't do that. Just use a virtualenv.

**Comment by yves-surrel** [comment]
I made a new conda environment, and simply did `conda install pyopencl`
I get the error `clGetPlatformIDs failed: PLATFORM_NOT_FOUND_KHR`in this new environment.

My fear of messing up system files was more about pocl, not PyOpenCL that I have been using successfully for years, thanks for that!

There may be something broken in the way the native MacOS OpenCL is found, because Apple does not do things as other vendors do, AFAIU.

**Comment by inducer** [comment]
Did you also `conda install pocl`?

**Comment by yves-surrel** [comment]
No, not in the new environment!

**Comment by yves-surrel** [comment]
Maybe the pocl uninstall in the previous environment (not the new one) is incomplete and it messes something at a global level?
Should we reinstall a full miniconda?

**Comment by inducer** [comment]
Yes, when in doubt, always start from a clean slate.

**Comment by isuruf** [comment, marked as answer]
You have to do `conda install ocl_icd_wrapper_apple` to bring back apple implementation
